CondFormats/EcalObjects/interface/EcalMultifitParametersSoA.h

```diff
@@ -0,0 +1,22 @@
+#ifndef CondFormats_EcalObjects_EcalMultifitParametersSoA_h
+#define CondFormats_EcalObjects_EcalMultifitParametersSoA_h
+
+#include <array>
+#include "DataFormats/SoATemplate/interface/SoACommon.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+#include "DataFormats/SoATemplate/interface/SoAView.h"
+
+constexpr size_t kNTimeFitParams = 8;
```

Comment:
Same here: are these number of parameters fixed? Shouldn't these numbers come from the algorithm that defines them?

Reply:
The parameters are taken from the CPU multifit configuration here: https://github.com/cms-sw/cmssw/blob/master/RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerMultiFit.cc#L744-L765
I do not know where they originally come from.
In the CPU ratio timing the parameters are stored in vectors and could theoretically change. I do not think this has ever happened though.

Reply:
Which of the two options is preferred? If it is the second, I can add the assert to the multifit code in this PR or a separate one.

Reply:
> Which of the two options is preferred? If it is the second, I can add the assert to the multifit code in this PR or a separate one.

Well, if it is feasible, I would go with the assert in the multifit code for all the sizes (re)defined here; better if it is in this PR, which had all signatures cleaned nonetheless.

Reply:
I would say that the check should go in the alpaka version of the multifit code, not in the "legacy" one that does not uses these conditions.

Reply:
If the check is to happen in the alpaka version then there is no change needed for this PR. The required assert will be part of the PR with the multifit migration.

It has already been implemented in the development branch https://github.com/thomreis/cmssw/commit/e0733d754b8149495d8961d0355932fe15261403

Reply:
> @perrotta are you happy with this resolution?

Ok, fine with me.
We can discuss it in your next development PR, but why do you check in some cases that the number of parameters is lower or equal than the EcalMultiFitConditionsSoA expects? Even if the code may technically work as such, shouldn't they be identically the same?

Reply:
I had a look at the tags on CondDB and the MC one only has one value for the barrel and one for the endcap. The data tag uses the sizes defined in the SoA.